semgrep/semgrep/semgrep_main.py

```diff
@@ -43,34 +43,25 @@ def validate_single_rule(config_id: str, rule: Dict[str, Any]) -> bool:
         print_error(
             f"{config_id} is missing required keys {missing_keys} at rule id {rule_id_err_msg}"
         )
-        return False
+        return None
     if not rule_keys.issubset(YAML_ALL_VALID_RULE_KEYS):
         extra_keys = rule_keys - YAML_ALL_VALID_RULE_KEYS
         print_error(
             f"{config_id} has invalid rule key {extra_keys} at rule id {rule_id_err_msg}, can only have: {YAML_ALL_VALID_RULE_KEYS}"
         )
-        return False
+        return None
     try:
-        _ = Rule.from_json(rule).expression
+        return Rule.from_json(rule)
     except InvalidRuleSchemaError as ex:
         print_error(
             f"{config_id}: inside rule id {rule_id_err_msg}, pattern fields can't look like this: {ex}"
         )
-        return False
-    try:
```

Comment:
this code doesn't do anything because looking up `globs` and `expressions` don't have side effects -- they're computed when the class is instantiated.